azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -319,27 +320,67 @@ public void updateProjectDescription(final Project project, final String descrip
     return this.projectLoader.getProjectEvents(project, results, skip);
   }
 
-  public Props getProperties(final Project project, final String source)
-      throws ProjectManagerException {
-    return this.projectLoader.fetchProjectProperty(project, source);
+  public Props getPropertiesFromFlowFile(final Flow flow, final String jobName, final String source,
+      final int flowVersion) throws ProjectManagerException {
+    final File flowFile = this.projectLoader.getUploadedFlowFile(flow.getProjectId(), flow
+        .getVersion(), flowVersion, source);
+    final String path =
+        jobName == null ? flow.getId() : flow.getId() + Constants.PATH_DELIMITER + jobName;
+    final Props props = FlowLoaderUtils.getPropsFromYamlFile(path, flowFile);
+    if (flowFile != null && flowFile.exists()) {
+      flowFile.delete();
+    }
+    return props;
   }
 
-  public Props getJobOverrideProperty(final Project project, final String jobName)
-      throws ProjectManagerException {
-    return this.projectLoader.fetchProjectProperty(project, jobName + ".jor");
+  public Props getProperties(final Project project, final Flow flow, final String jobName,
+      final String source) throws ProjectManagerException {
+    if (Double.compare(flow.getAzkabanFlowVersion(), Constants.AZKABAN_FLOW_VERSION_2_0) == 0) {
```

Comment:
consider to put the check in the util class like FlowUtils? i guess it will be used in multiple cases? but not every one will remember to use Double.compare to compare two doubles. At least for me it's not occasionally to just use "=="  :)

Reply:
Sure.